Landbosse cable connection from optiwindnet

jaredthomas68 opened the pull request:
This PR includes the ability to loosely tie LandBOSSE cable cost estimates to the cable design from optiwindnet by using the total optiwindnet cable length divided by the number of turbines as an estimate for the grid farm equivalent turbine spacing.

cfrontin commented on ard/cost/surrogate_turbine_spacing.py:
```diff
@@ -0,0 +1,87 @@
+from ard.cost.wisdem_wrap import LandBOSSE
+
+
+class LandBOSSEWithSurrogate(om.Group):
```
I'm slightly hesitant to use `Surrogate` here, since i associate it more with a low-order or reduced-order surrogate model. Now you might come back and say "well isn't this in some sense order reduction" and the answer is yes in some sense, but I think it might be worth it to save as a nice keyword to set aside for more primetime use later on.

jaredthomas68 replied:
Got it. I adjusted from "surrogate" to "approximations"

cfrontin commented on ard/cost/surrogate_turbine_spacing.py:
```diff
@@ -0,0 +1,87 @@
+        # Connect the turbine spacing output from the surrogate to LandBOSSE
+        self.connect("spacing_surrogate.primary_turbine_spacing_diameters", "internal_turbine_spacing_rotor_diameters")
+
+class PrimarySpacingSurrogate(om.ExplicitComponent):
```
I'm still a little wary to have a primary spacing surrogate without a secondary; I'm a little partial to having a zero-derivative, constant at say $7 D_{\\mathrm{rotor}}$ value dummy so that we aren't tempting people to accidentally cut it loose as a free design variable

jaredthomas68 replied:
I've added a secondary spacing that is calculated the same way as the primary spacing. My thought is that this will at least scale well. I've put it in in a way that I think will make it easy to change the estimation later if desired.

cfrontin reviewed (commented):
Looks pretty great. I added a few comments here and there. One other question I might add is: does this seem like a place to use a system test to make sure things are connecting well?

jaredthomas68 commented on ard/cost/surrogate_turbine_spacing.py:
```diff
@@ -0,0 +1,87 @@
+        outputs["primary_turbine_spacing_diameters"] = total_length_cables/(rotor_diameter_m*N_turbines)
+
+    def compute_partials(self, inputs, partials):
+        "partials are constant, so no calculations needed here"
```
This is the canonical way to handle constant partial derivatives in OpenMDAO

jaredthomas68 commented:
> Looks pretty great. I added a few comments here and there. One other question I might add is: does this seem like a place to use a system test to make sure things are connecting well?

I had a system test but had left it in the unit tests. I just moved it to the system tests.
